### ISSUE TITLE
compilation error on v0.4.2

### ISSUE THREAD

**Issue by defactojob** [gagliardetto/solana-go#4]
```
# github.com/gagliardetto/solana-go
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/transaction.go:260:2: undefined: bin.EncodeCompactU16Length
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/transaction.go:276:16: encoder.WriteBytes undefined (type *bin.Encoder has no field or method WriteBytes)
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/transaction.go:281:25: undefined: bin.DecodeCompactU16LengthFromByteReader
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/transaction.go:287:28: decoder.ReadNBytes undefined (type *bin.Decoder has no field or method ReadNBytes)
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/types.go:36:7: undefined: bin.EncoderDecoder
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/types.go:64:7: undefined: bin.EncoderDecoder
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/types.go:93:2: undefined: bin.EncodeCompactU16Length
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/types.go:100:2: undefined: bin.EncodeCompactU16Length
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/types.go:103:3: undefined: bin.EncodeCompactU16Length
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/types.go:108:3: undefined: bin.EncodeCompactU16Length
../../go/pkg/mod/github.com/gagliardetto/solana-go@v0.4.2/types.go:108:3: too many errors
```

**Comment by defactojob** [comment]
it's due to https://github.com/golang/go/issues/35450

> Exclude and replace apply only in the main module's go.mod and are ignored in dependencies.

you'd need to add to the top level project that imports solana-go

```
replace github.com/dfuse-io/binary => github.com/gagliardetto/binary v0.4.0
replace github.com/google/gofuzz => github.com/gagliardetto/gofuzz v1.2.1
```

**Comment by defactojob** [comment]
more compilation problem...

```
solana-go/rpc/ws/client.go:26:2: missing go.sum entry for module providing package github.com/buger/jsonparser (imported by github.com/gagliardetto/solana-go/rpc/ws); to add:
        go get github.com/gagliardetto/solana-go/rpc/ws@v1.0.0
solana-go/rpc/ws/client.go:27:2: missing go.sum entry for module providing package github.com/gorilla/rpc/v2/json2 (imported by github.com/gagliardetto/solana-go/rpc/ws); to add:
        go get github.com/gagliardetto/solana-go/rpc/ws@v1.0.0
solana-go/rpc/ws/client.go:28:2: missing go.sum entry for module providing package github.com/gorilla/websocket (imported by github.com/gagliardetto/solana-go/rpc/ws); to add:
        go get github.com/gagliardetto/solana-go/rpc/ws@v1.0.0
```

**Comment by gagliardetto** [comment]
What version of go are you compiling with?

**Comment by gagliardetto** [comment]
Hi! Thanks for filing an issue!

I fixed the problem and released `v0.4.3`

The correct dependencies now are:

```bash
go get github.com/gagliardetto/solana-go@v0.4.3
go get github.com/gagliardetto/binary@v0.4.1
go get github.com/gagliardetto/gofuzz@v1.2.2
```

**Comment by gagliardetto** [comment]
Closing.
